mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -188,6 +188,13 @@ CompletableFuture<PagedActiveJobOverview> getActiveJobOverview(
     CompletableFuture<Map<TaskExecutorID, WorkerId>> getTaskExecutorWorkerMapping(
         Map<String, String> attributes);
 
+    /**
+     * Signals that the given job id has been killed.
+     *
+     * @param jobId id of the job killed
+     */
+    CompletableFuture<Ack> unscheduleJob(String jobId);
```

Comment:
The interface seems to be leaking implementation details.

Reply:
As discussed in person, we can probably get rid of this and simply evict on cache TTL.